src/App.tsx

```diff
@@ -15,49 +17,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'NONE',
+  ALPHABET = 'ALPHABET',
+  LENGTH = 'LENGTH',
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
```

Comment:
```typescript
interface Person {
  name: string;
  age: number;
}

const person: Person = { name: 'Alice', age: 30 };
```

In TypeScript, the `interface` keyword is used to define object types. While the `type` keyword can also be used to define object types, `interface` has some advantages over `type` in certain cases. Here are some rules to consider when choosing between `interface` and `type` for typing objects in TypeScript:

1. **Prefer `interface` for object types:** The `interface` keyword is the preferred way to define object types in TypeScript. This is because interfaces can be extended, while types cannot. Interfaces can also be used to define callable objects, which is not possible with types.

2. **Use `type` for union types:** The `type` keyword is useful for defining union types, where a variable can have multiple possible types. This can be useful in cases where you want to define a variable that can either be a string or a number, for example.

3. **Use `interface` for extending:** If you need to extend an object type, you should use an interface. Interfaces can be extended with the `extends` keyword, which allows you to add or modify properties of an existing interface.

4. **Be consistent:** It's important to be consistent in your codebase when choosing between `interface` and `type` for object types. If you're working on a project with existing code, follow the established conventions. If you're starting a new project, choose one and stick with it.

In summary, while both `interface` and `type` can be used for typing objects in TypeScript, `interface` is the preferred way to define object types, especially when you need to extend an existing interface. The `type` keyword is useful for defining union types, but should be used sparingly for object types.